libs/libarchfpga/src/arch_util.cpp

```diff
@@ -944,7 +934,7 @@ void SyncModelsPbTypes_rec(t_arch* arch,
 
         pb_type->model_id = model_match_prim_id;
         vtr::t_linked_vptr* old = model_match_prim.pb_types;
-        model_match_prim.pb_types = (vtr::t_linked_vptr*)vtr::malloc(sizeof(vtr::t_linked_vptr));
+        model_match_prim.pb_types = new vtr::t_linked_vptr;
```

Comment:
We could probably change this linked list (which currently stores pb_type* pointers) to a vector of pb_type* (or maybe even just to direct pb_type elements).  Probably don't do that yet though ... we should refactor in stages.